app/models/g5_authenticatable/user.rb

```diff
@@ -38,7 +38,7 @@ def update_roles_from_auth(auth_data)
     end
 
     def clients
-      G5Updatable::ClientPolicy::Scope.new(self, G5Updatable::Client).resolve
+      G5Updatable::ClientPolicy::Scope.new(self, G5Updatable::Client).clients_from_client_and_location_roles
```

Comment:
I agree with @vgarro. It's probably best to encapsulate this logic in a separate policy with its own scope definition, via pundit's support for [headless policies](https://github.com/elabs/pundit#headless-policies). There's a place in the auth server where we're already doing something similar (we have a [`RolePolicy`](https://github.com/g5search/g5-authentication/blob/master/app/policies/role_policy.rb) and [`GlobalRolePolicy`](https://github.com/g5search/g5-authentication/blob/master/app/policies/global_role_policy.rb) that both deal with permissions on the `Role` model, but define different scopes for use in completely different contexts.

I think the name `IndirectClientThroughLocationPolicy` might be a tad too implementation-specific, though. Can we come up with something more abstract, but still descriptive that conveys intention? (e.g. `SelectableClientPolicy` or something like that) What do you guys think?